### PR TITLE
refactor: video tooltip should not close on hover events

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
## Context
Ticket: https://linear.app/prismic/issue/DT-1504/aauser-when-i-interact-out-side-of-the-video-tool-tip-it-should-not

Given I have a next project

[x] when have not seen the video tooltip or cleared local storage the video tooltip should open
[x] When I clicking out side of the tool-tip it should close
[x] when clicking the close button it should close
[x] when clicking the "got it" button it should close
[x] when hovering in and out should not close the tooltip
[x] Also ensure the close icon is properly aligned

## Checklist before requesting a review
- [ ] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.stories.tsx [hunk 1]
```diff
@@ -39,3 +39,23 @@ export const VideoPlayer = {
...
+  render: () => {
+    return (
+      <div>
+        <VideoPopover
+          open={true}
+          onClose={noop}
+          onPlay={noop}
+          cloudName="dmtf1daqp"
+          publicId="Tooltips/pa-course-overview_eaopsn"
+          delay={5000}
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/VideoPopover/VideoPopover.stories.tsx]: For the delay in storybook, for all stories I would put something really fast like 1 or 2 seconds, so we don't wait as long haha, WDYT?

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.test.tsx [hunk 2]
```diff
@@ -68,4 +68,74 @@ describe("VideoPopover", () => {
...
+          </VideoPopover>
+
+          <button>other</button>
+        </div>
+      );
+    };
+
+    const { user } = render(<App />);
+
+    const closeButton = await waitFor(() => screen.getByText("Got it"));
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/VideoPopover/VideoPopover.test.tsx]: As discussed:

```suggestion
    const closeButton = await screen.findByText("Got it");
```

And thanks to this, you could not have to do it in two lines:

```suggestion
    await user.hover(await screen.findByText("Got it"));
```

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx [hunk 3]
```diff
@@ -77,77 +159,24 @@ function useTimerWhen(condition: boolean, fn: () => void, delay: number) {
...
-  );
-
-  useTimerWhen(open, handleDefaultOpening, delay);
-
-  const handleChange = (value: boolean) => {
-    if (value === false) {
-      handleClose();
-    } else if (value === true && open === true) {
-      setOpen(value);
+function useOnClickOutside<T extends HTMLElement = HTMLElement>(
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: Following your question, yes, I would put this in our hooks folder.

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx [hunk 4]
```diff
@@ -77,77 +159,24 @@ function useTimerWhen(condition: boolean, fn: () => void, delay: number) {
...
+  ref: React.MutableRefObject<T | null>,
+  handler: () => void
+) {
+  useEffect(() => {
+    function checkWhereWasClicked(event: MouseEvent) {
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: `checkWhereWasClicked` is somewhat vague and not really what the function does.

I can propose `handleClickOutside` to be clearer, as it explains exactly when the function will be called, do you like it?

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx [hunk 5]
```diff
@@ -77,77 +159,24 @@ function useTimerWhen(condition: boolean, fn: () => void, delay: number) {
...
-        onClose={handleClose}
-        onPlay={onPlay}
-        cloudName={cloudName}
-        publicId={publicId}
-        thumbnail={thumbnail}
-      />
-    </HoverCard>
-  );
-};
+    document.addEventListener("mousedown", checkWhereWasClicked);
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: Maybe we should add and remote the event listenner only when the ref is not null?

```suggestion
    if (ref.current !== null) {
      document.addEventListener("mousedown", handleClickOutside);
    }

    return () => {
      if (ref.current !== null) {
        document.removeEventListener("mousedown", handleClickOutside);
      }
    }
```

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx [hunk 6]
```diff
@@ -77,77 +159,24 @@ function useTimerWhen(condition: boolean, fn: () => void, delay: number) {
...
-      handleClose();
-    } else if (value === true && open === true) {
-      setOpen(value);
+function useOnClickOutside<T extends HTMLElement = HTMLElement>(
+  ref: React.MutableRefObject<T | null>,
+  handler: () => void
+) {
+  useEffect(() => {
+    function checkWhereWasClicked(event: MouseEvent) {
+      if (
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: Best is not to use case if possible. Also, we can use optional chaining to make it shorter.

You could try something like:

```suggestion
     if (
        event.target instanceof Node &&
        ref.current?.contains(event.target) === false
      ) {
        handler();
      }
```

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.css.ts [hunk 7]
```diff
@@ -93,7 +93,11 @@ export const videoButton = style([
 
 export const closeButton = style([
   block,
-  sprinkles({ color: colors.purple12 }),
+  sprinkles({
+    color: colors.purple12,
+    padding: 6,
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/VideoPopover/VideoPopover.css.ts:98]: Not sure to understand, I see 4px as global padding and then 6px here so 10px before the icon, is that what we want?

**Review by xrutayisire — CHANGES_REQUESTED** [review]
Looks nice, no really big changes but small details I saw 🙂

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.css.ts (hunk 7 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/components/VideoPopover/VideoPopover.css.ts:98]: Depend which hover-card you look at: this one says 6 https://www.figma.com/file/wb7ujCJ4j7EiP7aDDGRYXB/SliceMachine%E2%84%A2-UI?node-id=4018%3A320396&mode=dev

but I can change it back to four and it shouldn't make much difference...

Comment (xrutayisire) [reply packages/slice-machine/src/components/VideoPopover/VideoPopover.css.ts:98]: Oh no that's perfect your link! Was not on that one, thank you!

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx (hunk 3 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: there's a similar hook in `usehooks-ts` ... not sure if it's worth inistalling the whole library for just a few hooks :/

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx (hunk 5 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: for removing it, i'd say it shouldn't matter if it's null or not (as the ref is mutablle)
for addding yes, that makes sense :)

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.stories.tsx (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/components/VideoPopover/VideoPopover.stories.tsx]: 1.5

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx (hunk 4 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: can do

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx (hunk 3 above)
Comment (xrutayisire) [reply packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: No don't think so, putting it in hooks folder is great already for me

File: packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx (hunk 5 above)
Comment (xrutayisire) [reply packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx]: Good one! 100% agree

**Review by xrutayisire — APPROVED** [review]
LGTM 🙂

**Pull request merged**